Source/Core/Core/PowerPC/JitArm64/JitArm64_Integer.cpp

```diff
@@ -1359,16 +1382,68 @@ void JitArm64::srawx(UGeckoInstruction inst)
       ComputeRC0(gpr.GetImm(a));
     return;
   }
-
-  if (gpr.IsImm(b) && !js.op->wantsCA)
+  else if (gpr.IsImm(s) && gpr.GetImm(s) == 0)
+  {
+    gpr.SetImmediate(a, 0);
+    ComputeCarry(false);
+    if (inst.Rc)
+      ComputeRC0(0);
+    return;
+  }
+  else if (gpr.IsImm(b))
   {
     int amount = gpr.GetImm(b);
-    if (amount & 0x20)
-      amount = 0x1F;
+
+    bool special = amount & 0x20;
+    amount &= 0x1f;
+
+    if (special)
+    {
+      gpr.BindToRegister(a, a == s);
+
+      ASR(gpr.R(a), gpr.R(s), 31);
+
+      if (js.op->wantsCA)
+      {
+        CMN(gpr.R(s), gpr.R(s));
+        ComputeCarry();
+      }
+    }
+    else if (amount == 0)
+    {
+      if (a != s)
+      {
+        gpr.BindToRegister(a, false);
+
+        ADDS(gpr.R(a), gpr.R(s), 0);
+        ComputeCarry();
```

Comment:
Carry is always false here. So you can use MOV.

Reply:
I figured this was an efficient way to set the carry flag in case the next instruction wants it in the host carry flag. But I should probably make it use MOV instead when `!js.op->wantsCA`.

Reply:
We discussed this briefly on IRC the day the comment was made, and the best solution long-term is probably to implement a way for the following instruction to read the carry flag at compile-time. So I've made this use `MOV` plus `ComputeCarry(false)` in all cases.